ably/rest_http_error.go

```diff
@@ -6,15 +6,16 @@ import (
 )
 
 type RestHttpError struct {
-	Msg      string
-	Response *http.Response
-
+	Msg          string
 	ResponseBody string
+
+	resp *http.Response
```

Comment:
It feels odd to have an HTTP Error without the status code as part of the type?  Also, note that our responses always follow a generic format in JSON, see https://rest.ably.io/ as an example.  We therefore have an Ably ErrorInfo type, see https://github.com/ably/ably-java/blob/master/src/io/ably/types/ErrorInfo.java.  Should we not follow that format?